package/MDAnalysis/coordinates/base.py

```diff
@@ -1082,54 +1162,70 @@ def _sliced_iter(self, start, stop, step):
         """
         # override with an appropriate implementation e.g. using self[i] might
         # be much slower than skipping steps in a next() loop
-        def _iter(start=start, stop=stop, step=step):
-            try:
-                for i in xrange(start, stop, step):
-                    yield self._read_frame(i)
-            except TypeError:  # if _read_frame not implemented
-                raise TypeError("{0} does not support slicing."
-                                "".format(self.__class__.__name__))
-        return _iter()
-
-    def _check_slice_indices(self, start, stop, step):
-        """Unpack the slice object and do checks
+        try:
+            for i in xrange(start, stop, step):
+                yield self._read_frame(i)
+        except TypeError:  # if _read_frame not implemented
+            raise TypeError("{0} does not support slicing."
+                            "".format(self.__class__.__name__))
+
+    def check_slice_indices(self, start, stop, step):
+        """Check frame indices are valid and clip to fit trajectory
+
+        Parameters
+        ----------
+        start, stop, step
```

Comment:
I think it makes sense to group arguments that share types, and it's allowed:

When two or more input parameters have exactly the same type, shape and description, they can be combined:

x1, x2 : array_like
    Input arrays, description of `x1`, `x2`.